src/code/z_eff_blure.c

```diff
@@ -66,40 +68,977 @@ void func_800A81F0(EffectBlure* this, Vec3f* p1, Vec3f* p2) {
+        vtx[0].v.cn[0] = sp78.r;
+        vtx[0].v.cn[1] = sp78.g;
+        vtx[0].v.cn[2] = sp78.b;
+        vtx[0].v.cn[3] = sp78.a;
+
+        sp60.x = sp8C.x;
+        sp60.y = sp8C.y;
+        sp60.z = sp8C.z;
+        Math_Vec3f_Diff(&sp60, &sp6C, &sp54);
+        Math_Vec3f_Scale(&sp54, 10.0f);
+        vtx[1].v.ob[0] = sp54.x;
+        vtx[1].v.ob[1] = sp54.y;
+        vtx[1].v.ob[2] = sp54.z;
+        vtx[1].v.cn[0] = sp7C.r;
+        vtx[1].v.cn[1] = sp7C.g;
+        vtx[1].v.cn[2] = sp7C.b;
+        vtx[1].v.cn[3] = sp7C.a;
+
+        ratio = (f32)(elem + 1)->timer / (f32)this->elemDuration;
+        EffectBlure_GetComputedValues(this, index + 1, ratio, &sp8C, &sp84, &sp7C, &sp78);
+
+        sp60.x = sp8C.x;
+        sp60.y = sp8C.y;
+        sp60.z = sp8C.z;
+        Math_Vec3f_Diff(&sp60, &sp6C, &sp54);
+        Math_Vec3f_Scale(&sp54, 10.0f);
+        vtx[2].v.ob[0] = sp54.x;
+        vtx[2].v.ob[1] = sp54.y;
+        vtx[2].v.ob[2] = sp54.z;
+        vtx[2].v.cn[0] = sp7C.r;
+        vtx[2].v.cn[1] = sp7C.g;
+        vtx[2].v.cn[2] = sp7C.b;
+        vtx[2].v.cn[3] = sp7C.a;
+
+        sp60.x = sp84.x;
+        sp60.y = sp84.y;
+        sp60.z = sp84.z;
+        Math_Vec3f_Diff(&sp60, &sp6C, &sp54);
+        Math_Vec3f_Scale(&sp54, 10.0f);
+        vtx[3].v.ob[0] = sp54.x;
+        vtx[3].v.ob[1] = sp54.y;
+        vtx[3].v.ob[2] = sp54.z;
+        vtx[3].v.cn[0] = sp78.r;
+        vtx[3].v.cn[1] = sp78.g;
+        vtx[3].v.cn[2] = sp78.b;
+        vtx[3].v.cn[3] = sp78.a;
+
+        gSPVertex(POLY_XLU_DISP++, vtx, 4, 0);
+        gSP2Triangles(POLY_XLU_DISP++, 0, 1, 2, 0, 0, 2, 3, 0);
+    }
+
+    CLOSE_DISPS(gfxCtx);
+}
+
+void EffectBlure_DrawElemHermiteInterpolation(EffectBlure* this, EffectBlureElement* elem, s32 index,
+                                              GraphicsContext* gfxCtx) {
+    static Vtx_t baseVtx = VTX_T(0, 0, 0, 0, 0, 0xFF, 0xFF, 0xFF, 0xFF);
+    Vtx* vtx;
+    Vec3s sp1EC;
+    Vec3s sp1E4;
+    f32 ratio;
+    Color_RGBA8 sp1DC;
+    Color_RGBA8 sp1D8;
+    Vec3f sp1CC;
+    Vec3f sp1C0;
+    Vec3f sp1B4;
+    Vec3f sp1A8;
+    Color_RGBA8 sp1A4;
+    Color_RGBA8 sp1A0;
+    Color_RGBA8 sp19C;
+    Color_RGBA8 sp198;
+    Vec3f sp18C;
+    Vec3f sp180;
+    Vec3f sp174;
+    Vec3f sp168;
+    s32 i;
+    Vec3f sp158;
+    Vec3f sp14C;
+    Color_RGBA8 sp148;
+    Color_RGBA8 sp144;
+    Vec3f sp138;
+
+    OPEN_DISPS(gfxCtx);
+
+    Math_Vec3s_ToVec3f(&sp138, &this->elements[0].p2);
+
+    ratio = (f32)elem->timer / (f32)this->elemDuration;
+    EffectBlure_GetComputedValues(this, index, ratio, &sp1EC, &sp1E4, &sp1A4, &sp1A0);
+    Math_Vec3s_ToVec3f(&sp1CC, &sp1EC);
+    Math_Vec3s_ToVec3f(&sp1C0, &sp1E4);
+
+    ratio = (f32)(elem + 1)->timer / (f32)this->elemDuration;
+    EffectBlure_GetComputedValues(this, index + 1, ratio, &sp1EC, &sp1E4, &sp19C, &sp198);
+    Math_Vec3s_ToVec3f(&sp18C, &sp1EC);
+    Math_Vec3s_ToVec3f(&sp180, &sp1E4);
+
+    if ((elem->flags & 3) == 2) {
```

Comment:
```suggestion
    if ((elem->flags & (2 | 1)) == 2) {
```
imo this is easier to read since it's clearer that it's checking to make sure both bit 2 is on and bit 1 is off, but I could understand if the brackets look a little funny